server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -383,7 +383,8 @@ private SearchHits getHits(ReducedQueryPhase reducedQueryPhase, boolean ignoreFr
             }
         }
         return new SearchHits(hits.toArray(new SearchHit[0]), reducedQueryPhase.totalHits,
-            reducedQueryPhase.maxScore, sortedTopDocs.sortFields, sortedTopDocs.collapseField, sortedTopDocs.collapseValues);
+            hits.isEmpty() ? Float.NaN : reducedQueryPhase.maxScore,
```

Comment:
I am not sure that this is right, we should rather track the root cause for this. It gets quite complicated though as there are places where we do have empty hits and maxScore set to some value. I've seen quite some scroll tests fail due to this, as well as others.